hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/MockAbfsInputStream.java

```diff
@@ -125,9 +147,82 @@ public void induceConnectionException() {
+    double session_refresh_internal_factor = 0.75;
+    ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
+    ReentrantReadWriteLock rwLock = new ReentrantReadWriteLock();
+
+    // override fields
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "LOG", log);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "SESSION_REFRESH_INTERVAL_FACTOR", session_refresh_internal_factor);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "client", client);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "path", path);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "eTag", eTag);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "tracingContext", tracingContext);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "scheduledExecutorService", scheduledExecutorService);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "rwLock", rwLock);
+
+    doCallRealMethod().when(mockSession)
+        .updateAbfsFastpathSessionToken(any(), any());
+    doCallRealMethod().when(mockSession)
+        .updateConnectionMode(any(AbfsConnectionMode.class));
+    doCallRealMethod().when(mockSession)
+        .updateConnectionModeForFailures(any(AbfsConnectionMode.class));
+    doCallRealMethod().when(mockSession).close();
+    doCallRealMethod().when(mockSession)
+        .setAbfsFastpathSessionInfo(any(AbfsFastpathSessionInfo.class));
+    doCallRealMethod().when(mockSession)
+        .setConnectionMode(any(AbfsConnectionMode.class));
+
+    when(mockSession.executeFastpathClose()).thenCallRealMethod();
+    when(mockSession.executeFastpathOpen()).thenCallRealMethod();
+    when(mockSession.getCurrentAbfsFastpathSessionInfoCopy()).thenCallRealMethod();
+    when(mockSession.executeFetchFastpathSessionToken()).thenCallRealMethod();
+    when(mockSession.getSessionRefreshIntervalInSec()).thenCallRealMethod();
+    when(mockSession.fetchFastpathSessionToken()).thenCallRealMethod();
+
+    return mockSession;
+  }
+
+  public static AbfsRestOperation getMockSuccessRestOp(AbfsClient client, byte[] token, Duration tokenDuration)
+      throws IOException {
+    AbfsRestOperation op = mock(AbfsRestOperation.class);
+    AbfsHttpOperation httpOp = mock(AbfsHttpOperation.class);
+    AbfsConfiguration abfsConfig = client.getAbfsConfiguration();
+    byte[] accountKey = Base64.decode(abfsConfig.getStorageAccountKey());
+    ServiceSASGenerator sasGenerator = new ServiceSASGenerator(accountKey);
+    String se = ISO_8601_FORMATTER.format(java.time.Instant.now().plus(tokenDuration));
+    System.out.println("New Expiry - " + se);
```

Comment:
is the system.out.println for aiding debugging?

Reply:
Log not needed. Removing.